tests/stack-test.cc

```diff
@@ -72,8 +72,42 @@ namespace
 		debug_log("Expected to invoke the handler? {}", handlerExpected);
 		set_expected_behaviour(&threadStackTestFailed, handlerExpected);
 	}
+
+	__attribute__((used)) extern "C" int test_small_stack()
+	{
+		return test_stack_requirement();
+	}
 } // namespace
 
+__attribute__((used)) extern "C" int test_with_small_stack(size_t stackSize);
+
+asm(".section .text\n"
+    ".global test_with_small_stack\n"
+    "test_with_small_stack:\n"
+    // Preserve the old stack
+    "  cmove ct0, csp\n"
+    // Add space to the requested size for the spill slots and the size of the
+    // stack that `test_small_stack` will use, plus the four capabilities that
+    // the switcher will spill for us.
+    "  add a0, a0, 64\n"
```

Comment:
I guess this is 2 x spills here, 2 x spills in test_small_stack and 4 x switcher spills? Would be good to be explicit. Are the spills in `test_small_stack` dependent on anything like debug level? Concerned the magic number may be a bit brittle.

Reply:
The magic number *is* brittle, but the stack must be 16-byte aligned and we check 16 bytes either side of the line, so we'll at least get a failure that someone can debug if one of the assumptions changes.  I'll improve the comment.